client/src/controllers/TabsController.ts

```diff
@@ -0,0 +1,280 @@
+import { Controller } from '@hotwired/stimulus';
+
+const TAB_DEFAULTS = {
+  css: {
+    animate: 'animate-in',
+  },
+  keys: {
+    end: 'End',
+    home: 'Home',
+    left: 'ArrowLeft',
+    up: 'ArrowUp',
+    right: 'ArrowRight',
+    down: 'ArrowDown',
+  },
+  direction: {
+    ArrowLeft: -1,
+    ArrowRight: 1,
+  },
+}
+
+export class TabsController extends Controller<HTMLDivElement> {
+  static targets = ['tabList', 'tabTrigger', 'tabLabel', 'tabPanel'];
+
+  static values = {
+    transition: { default: 150, type: Number },
+    activeTabId: { default: '', type: String },
+    disableURL: { default: false, type: Boolean },
+    initialPageLoad: { default: true, type: Boolean },
+    animate: { default: true, type: Boolean },
+  };
+
+  declare tabListTarget: HTMLDivElement;
+  declare tabTriggerTargets: HTMLElement[];
+  declare tabLabelTargets: HTMLAnchorElement[];
+  declare tabPanelTargets: HTMLElement[];
+
+  declare disableURLValue: boolean;
+  declare activeTabIdValue: string;
+  declare transitionValue: number;
+  declare animateValue: boolean;
+
+  initialize() {
+    if (this.tabTriggerTargets) {
```

Comment:
No need to check the if here. Targets will always return an array so this will always be truthy.